ckanext/cnra_schema/plugins.py

```diff
@@ -26,6 +29,19 @@ def update_config(self, config):
 """
 
     def get_package_dict(self, context, data_dict):
+        harvest_object = data_dict['harvest_object']
+        harvest_source_type = harvest_object.source.type
+
+        if harvest_source_type == 'fgdc':
+            modified_package_dict = self.get_fgdc_package_dict(data_dict)
+
+        elif harvest_source_type == 'waf':
+            modified_package_dict = self.get_waf_package_dict(data_dict)
+        
```

Comment:
If `harvest_source_type` is not either fgdc or waf then `modified_package_dict` will be undefined.
You should return `data_dict['package_dict']` if `harvest_source_type` in not fgdc or waf.